internal/services/hsm/dedicated_hardware_security_module_resource_test.go

```diff
@@ -15,7 +15,28 @@ import (
 
 type DedicatedHardwareSecurityModuleResource struct{}
 
-func TestAccDedicatedHardwareSecurityModule_basic(t *testing.T) {
+func TestAccDedicatedHardwareSecurityModule(t *testing.T) {
+	testCases := map[string]map[string]func(t *testing.T){
+		"Resource": {
+			"basic":          testAccDedicatedHardwareSecurityModule_basic,
+			"requiresImport": testAccDedicatedHardwareSecurityModule_requiresImport,
+			"complete":       testAccDedicatedHardwareSecurityModule_complete,
+			"update":         testAccDedicatedHardwareSecurityModule_update,
+		},
+	}
```

Comment:
Could you outline the reason for this?

Reply:
HSM has low quota and potentially slow recycle time so that these TCs failed. So made this PR to run them in sequence.

Reply:
Per the [settings](https://github.com/hashicorp/terraform-provider-azurerm/blob/162634977ee26d92e4c2ba5ccebc774194a29a95/.teamcity/components/settings.kt#L76), seems the TCs are already in sequence.

Reply:
yes please, we try to always use teamcity if we can to work around quota issues. increasing quota is likely the solution here

Reply:
Or, is it that the resource is returning before the resource is actually deleted/free'd up?

Reply:
It returned after LRO is completed.